bundles/org.eclipse.osgi/container/src/org/eclipse/osgi/storage/url/BundleURLConnection.java

```diff
@@ -131,7 +131,8 @@ public long getLastModified() {
 	 * @return the local URL using a common local protocol
 	 */
 	public URL getLocalURL() {
-		return bundleEntry.getLocalURL();
+		URL local = bundleEntry.getLocalURL();
+		return local == null ? url : local;
```

Comment:
```suggestion
		return local == null ? getURL() : local;
```

Wouldn't it be better to use the accessor method rather than a field reference? :-)

Reply:
Its a protected field from super that is accessed directly from other code in this class.  I would like to stay consistent so perhaps use `getURL` in all locations `url` is accessed here?